### PR TITLE
OSOE-606: Upgrade to C# 11

### LINKED ISSUE #84: Upgrade to C# 11 (OSOE-606)
See https://github.com/Lombiq/.NET-Analyzers#upgrading-to-a-new-version-of-the-net-sdk

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-606)

### PR DESCRIPTION
**Pull request by Piedone**
[OSOE-606](https://lombiq.atlassian.net/browse/OSOE-606)
Fixes #84

[OSOE-606]: https://lombiq.atlassian.net/browse/OSOE-606?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.Analyzers/Build.props [hunk 1]
```diff
@@ -2,21 +2,21 @@
   <PropertyGroup>
     <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)orchardcore.ruleset</CodeAnalysisRuleSet>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <!-- Don't run analyzers during Visual Studio build, not to slow down development. They still run during dotnet build.
-         By making this conditional, it won't overwrite the value if you set it in the Directory.Build.props file. For
-         Rider users: https://youtrack.jetbrains.com/issue/RIDER-80579 -->
     <RunAnalyzersDuringBuild Condition="'$(RunAnalyzersDuringBuild)' != 'true'">false</RunAnalyzersDuringBuild>
+    <!-- Don't run analyzers during Visual Studio build, not to slow down development. They still run during dotnet
+         build. By making this conditional, it won't overwrite the value if you set it in the Directory.Build.props
+         file. For Rider users: https://youtrack.jetbrains.com/issue/RIDER-80579 -->
```
Comment (0liver) [inline Lombiq.Analyzers/Build.props:7]: Why has this comment moved below the respective property?

File: Lombiq.Analyzers/general.ruleset [hunk 2]
```diff
@@ -386,7 +386,6 @@
     <Rule Id="S3906" Action="Warning" />
     <Rule Id="S3908" Action="Warning" />
     <Rule Id="S3937" Action="Warning" />
-    <Rule Id="S3949" Action="Warning" />
```
Comment (0liver) [inline Lombiq.Analyzers/general.ruleset]: Why were these removed?

File: Lombiq.Analyzers/Lombiq.Analyzers.csproj [hunk 3]
```diff
@@ -3,6 +3,9 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <DefaultItemExcludes>$(DefaultItemExcludes);.git*;</DefaultItemExcludes>
+    <!-- This is duplicated with Lombiq.Analyzers.nuspec.template but is needed for the NuGet publishing workflow not to
+    throw a warning. -->
+    <PackageLicenseFile>License.md</PackageLicenseFile>
```
Comment (0liver) [inline Lombiq.Analyzers/Lombiq.Analyzers.csproj]: Did you report that problem to the NuGet project?

**Review by 0liver — CHANGES_REQUESTED** [review]

File: Lombiq.Analyzers/Lombiq.Analyzers.csproj (hunk 3 above)
Comment (Piedone) [reply Lombiq.Analyzers/Lombiq.Analyzers.csproj]: It's not a NuGet issue but ours, and I did add one for it: https://github.com/Lombiq/GitHub-Actions/issues/236. I also linked it now.

File: Lombiq.Analyzers/Build.props (hunk 1 above)
Comment (Piedone) [reply Lombiq.Analyzers/Build.props:7]: It shouldn't, a mistake.

File: Lombiq.Analyzers/general.ruleset (hunk 2 above)
Comment (Piedone) [reply Lombiq.Analyzers/general.ruleset]: These were removed by the VS editor automatically. This happens when rules are removed, or their defaults change. However, I just checked and this is not the case for these rules, so I'll re-add them.

**Review by 0liver — APPROVED** [review]

File: Readme.md [hunk 4]
```diff
@@ -46,6 +46,10 @@ Bug reports, feature requests, comments, questions, code contributions and love
 
 This project is developed by [Lombiq Technologies](https://lombiq.com/). Commercial-grade support is available through Lombiq.
 
+### Publishing a new NuGet package
+
+When publishing a new version of the NuGet package, even if it's just a prerelease, also update the package references under _TestSolutions\Lombiq.Analyzers.PackageReferences_ to the new version. This is necessary to run the tests there
```
Comment (0liver) [inline Readme.md]:
```suggestion
When publishing a new version of the NuGet package, even if it's just a pre-release, also update the package references under _TestSolutions\Lombiq.Analyzers.PackageReferences_ to the new version. This is necessary to run the tests there.
```

**Review by 0liver — CHANGES_REQUESTED** [review]
